lib/index.js

```diff
@@ -1,167 +1,169 @@
+   * defined in the `connect` function provided.
+   *
+   * @constructor
+   * @type {module.Datastar}
+   * @param {Object} options - Config options options
+   * @param {function} connect -
+   */
+  constructor(options, connect) {
+    this.options = options || {};
+    this.connect = connect || this.connect;
+  }
 
-//
-// Alias to joi for defining schema for a model
-//
-Datastar.prototype.schema = jtc;
+  /**
+   * Attach the connection to the model constructor
+   * @param {Object} objectModel - Defined object model
+   * @returns {Datastar} - The datastar object with the attached object model
+   */
+  attach(objectModel) {
+    if (!this.connection) {
+      this.connect();
+    }
 
-//
-// Expose StatementCollection on the datastar instance
-//
-Datastar.prototype.StatementCollection = require('./statement-collection');
+    objectModel.connection = this.connection;
 
-/**
- * Attach the connection to the model constructor
- * @param {Object} objectModel - Defined object model
- * @returns {Datastar} - The datastar object with the attached object model
- */
-Datastar.prototype.attach = function attach(objectModel) {
-  if (!this.connection) {
-    this.connect();
+    return this;
   }
 
-  objectModel.connection = this.connection;
-
-  return this;
-};
-
-/**
- * Default connection logic which works with `priam`. This abstract
- * exists for future extensibility.
- * @param {Function} callback - The async callback
- * @returns {Datastar} - The datastar object
- */
-Datastar.prototype.connect = function connect(callback) {
-  var config = this.options.config;
-  //
-  // Use the cached connection if a model has been defined already via `attach`
-  // so we dont create more than 1 priam instance. This allows `connect` to be
-  // called with the callback to ensure the connection is pre-heated for all
-  // models
-  //
-  this.connection = this.connection || new Priam(this.options);
-
-  var create = config && config.keyspaceOptions &&
-      'CREATE KEYSPACE IF NOT EXISTS ' + config.keyspace +
-      ' WITH replication = ' +
-      JSON.stringify(config.keyspaceOptions).replace(/"/g, "'") +
-      ';';
-
-  if (create) {
+  /**
+   * Default connection logic which works with `priam`. This abstract
+   * exists for future extensibility.
+   * @param {Function} callback - The async callback
+   * @returns {Datastar} - The datastar object
+   */
+  connect(callback) {
+    const config = this.options.config;
+    //
+    // Use the cached connection if a model has been defined already via `attach`
+    // so we dont create more than 1 priam instance. This allows `connect` to be
+    // called with the callback to ensure the connection is pre-heated for all
+    // models
     //
-    // Try to create the keyspace. As a side effect pre-heat the connection
+    this.connection = this.connection || new Priam(this.options);
+
+    const create = config && config.keyspaceOptions &&
+        'CREATE KEYSPACE IF NOT EXISTS ' + config.keyspace +
```

Comment:
Can template string all the things if you want, but I'm fine with this also since it reads nicely.

Reply:
@scommisso: replaced most string concatenations with template strings.